collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -84,14 +87,17 @@ public void doScrape() throws Exception {
         if (jmxUrl.isEmpty()) {
           beanConn = ManagementFactory.getPlatformMBeanServer();
         } else {
-          HashMap credential = null;
+          HashMap<String, Object> env = new HashMap<String, Object>();
           if(username != null && username.length() != 0 && password != null && password.length() != 0) {
-            credential = new HashMap();
             String[] credent = new String[] {username, password};
-            credential.put(javax.management.remote.JMXConnector.CREDENTIALS, credent);
+            env.put(javax.management.remote.JMXConnector.CREDENTIALS, credent);
           }       
 
-          jmxc = JMXConnectorFactory.connect(new JMXServiceURL(jmxUrl), credential);
+          SslRMIClientSocketFactory clientSocketFactory = new SslRMIClientSocketFactory();
+          env.put(Context.SECURITY_PROTOCOL, "ssl");
```

Comment:
This looks like it requires all users to use SSL, which wouldn't work out too well.